CMakeLists.txt

```diff
@@ -0,0 +1,59 @@
+cmake_minimum_required(VERSION 2.8)
+
+include(GNUInstallDirs)
+
+find_package(LLVM REQUIRED CONFIG)
```

Comment:
The way we find LLVM needs to be encapsulated in a function so we can support multiple ways of doing this. Grabbing via CMake was broken for quite a while in Ubuntu IIRC.

As a fallback we should invoke the `llvm-config` executable to get this information.